spyder/plugins/maininterpreter.py

```diff
@@ -222,5 +230,20 @@ def set_umr_namelist(self):
                 fixed_namelist = []
             self.set_option('umr/namelist', fixed_namelist)
 
+    def set_custom_interpreters_list(self, executable):
+        """Update the list of interpreters used and the current one."""
+        custom_list = self.get_option('custom_list')

```

Comment:
We need to add a method that takes this list and verifies the entries that still exist (with a simple `osp.isfile`). The ones that don't need to be removed and a new list of interpreters needs to be saved with `set_option`.

Reply:
Sorry, this probably doesn't need to be done here, but above, where you create the filecombobox.